src/jit/utils.cpp

```diff
@@ -1954,6 +1954,42 @@ float FloatingPointUtils::round(float x)
     return _copysignf(flrTempVal, x);
 }
 
+bool FloatingPointUtils::isNormal(double x)
+{
+    int64_t bits = *(int64_t*)&x;
+    bits &= 0x7FFFFFFFFFFFFFFF;
+    return (bits < 0x7FF0000000000000) && (bits != 0) && ((bits & 0x7FF0000000000000) != 0);
+}
+
+bool FloatingPointUtils::isNormal(float x)
+{
+    int32_t bits = *(int32_t*)&x;
+    bits &= 0x7FFFFFFF;
+    return (bits < 0x7F800000) && (bits != 0) && ((bits & 0x7F800000) != 0);
+}
+
+bool FloatingPointUtils::isPow2(double x)
```

Comment:
I liked mikedn's suggestion of `hasPreciseReciprocal` since we aren't doing proper `isPow2` check, it also is more similar to the `hasExactInverseFP` name that clang uses.